static_provider_test.go

```diff
@@ -271,3 +272,26 @@ func TestPopulateForMapsWithNotAssignableKeyTypes(t *testing.T) {
 	assert.Contains(t, msg, `can't convert "1" to "config.secretType"`)
 	assert.Contains(t, msg, "key types conversion")
 }
+
+func TestValue_ChildKeys(t *testing.T) {
+	t.Parallel()
+
+	p := NewStaticProvider(map[string]interface{}{
+		"one":   1,
+		"two":   2,
+		"slice": []int{42, 13},
+	})
+
+	op := func(t *testing.T, key string, expected []string) {
+		keys := p.Get(key).ChildKeys()
+		sort.Strings(keys)
+		assert.Equal(t, expected, keys)
+	}
+
+	t.Run("Map", func(t *testing.T) { op(t, Root, []string{"one", "slice", "two"}) })
+	t.Run("Slice", func(t *testing.T) { op(t, "slice", []string{"13", "42"}) })
```

Comment:
Well, I was thinking about it as a collection of elements:
```yaml
- 13
- 42
```
but you right, the go way, would be {0, 1}